### PR TITLE
first test and circleci config

### PR DESCRIPTION
**Pull request by benadida**

### CODE REVIEW

File: client/src/App.test.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+import { fireEvent, render, screen } from '@testing-library/react'
+import fakeKiosk from '../test/helpers/fakeKiosk'
+
+import App from './App'
+
+beforeEach(() => {
+  // we don't care about network errors logged to the console, they're crowding things
+  jest.spyOn(console, 'error').mockImplementation(() => {}) // eslint-disable-line @typescript-eslint/no-empty-function
```
Comment (eventualbuddha) [inline client/src/App.test.tsx:9]: This is a bit of a code smell. Can we stop them from logging in the first place?

File: client/src/App.test.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  window.location.href = '/'
+  window.kiosk = fakeKiosk()
+})
+
+it('basic navigation works', async () => {
+  const { getByText } = render(<App />)
+
+  await screen.findByText('Create New Election Definition')
```
Comment (eventualbuddha) [inline client/src/App.test.tsx:18]: Why is this needed?

File: client/src/App.test.tsx (hunk 1 above)
Comment (benadida) [reply client/src/App.test.tsx:9]: we cannot stop it from logging the network error. Note that it doesn't cause the test to fail when the network error occurs, and because the network error *is* expected in this case (it's normal system operation), this feels like the best way to not make running tests look unnecessarily scary.

File: client/src/App.test.tsx (hunk 2 above)
Comment (benadida) [reply client/src/App.test.tsx:18]: Because there is enough asynchronous stuff going on (including that network call that fails) that if we don't wait for this, we fail tests.
